tests/certification/state/azure/cosmosdb/README.md

```diff
@@ -0,0 +1,18 @@
+# Azure Cosmos DB certification testing
+
+This project aims to test the Azure Cosmos DB State Store component under various conditions.
+
+## Test plan
+
+### Basic Test using master key authentication
+1. Able to create and test connection.
+2. Able to do set, fetch and delete.
+
+### Test Partition Keys
```

Comment:
Please add a test where you are connecting to a different collection -- a collection which does not use the expected `/partitionKey` partition key. We want to assert the failure for that case. This is a common problem.

To do so, please add a new collection with wrong partition name here:
https://github.com/dapr/components-contrib/blob/master/.github/infrastructure/conformance/azure/conf-test-azure-cosmosdb.bicep#L44

Then also make sure to update the following so that the second collection name gets exported to the KeyVault where we save these configurations
https://github.com/dapr/components-contrib/blob/master/.github/infrastructure/conformance/azure/conf-test-azure.bicep
https://github.com/dapr/components-contrib/blob/master/.github/infrastructure/conformance/azure/setup-azure-conf-test.sh

Lastly, set up a new component YAML talking to the collection with the bad partition name. And make sure to reference the new environment variable you are creating above.

Don't forget to inject that environment variable into the workflow run via the required-secrets section!

Reply:
To provide more context -- what you have described is only impacting partition key values, but the Cosmos DB state store requires that the Cosmos DB collection was setup with a partition Key that is exactly called `/partitionKey`. It will be good to assert the failure when the collection has been set up wrong.